pylint/plugins/hass_imports.py

```diff
@@ -346,6 +356,11 @@ def visit_importfrom(self, node: nodes.ImportFrom) -> None:
                 if name[0] == self.current_package.split(".")[2]:
                     self.add_message("hass-relative-import", node=node)
             return
+        if node.modname.startswith("homeassistant.components.") and (
+            node.modname.endswith(".const") or "const" in node.names
```

Comment:
`node.names` is a tuple. Thus checking for `"const" in node.names` probably doesn't work.

Reply:
I've implemented a fix for this.
It highlighted new components that needed fixing but those are all in now.